File: 000094.py
#!/usr/bin/env python
# -*- coding: utf-8 -*-
#
# 000094.py
#

Sum = 0
c = 3
Abort = False
while not Abort:
    for z in [-1, 1]:
        if Abort:
            continue
        B = c + z
        b = B//2
        a = (c**2 - b**2)**.5
        if (B*a) % 3 == 0:
            L = 2*c + B
            # print(int(a), b, c, B, L)
            if (L > 1000000):
                Abort = True
                break
            Sum += L
    c += 2

print('ANSWER: ', Sum)

File: 000094.pas
#!/usr/bin/env instantfpc

{$mode objfpc}{$H+}
{$APPTYPE CONSOLE}

uses
  Crt;

var
  a, b, c: Int64;
  ar: Extended;
  Base: Integer;
  Sum: Integer;
  z, L: Integer;
  Abort: Boolean;
begin
  ClrScr;
  c := 3;
  Sum := 0;
  Abort:= False;
  while not Abort do begin
    for z := -1 to 1 do begin
      if Abort then Break;
      if z = 0 then Continue;
      Base := c + z;
      b := Base div 2;
      //WriteLn(c, ' ', b);
      ar:= Sqrt(c * c - b * b);
      a := Trunc(ar);
      if Frac(ar) = 0 then
      if (Base * a) mod 3 = 0 then begin
        L := 2 * c + Base;
        //WriteLn(ar, ' ', a, ' ',b, ' ', c, ' ', Base, ' ', L);
        if L > 1000000000 then begin
          Abort := True;
          Break;
        end;
        Inc(Sum, L);
      end;
    end;
    Inc(c, 2);
  end;
  WriteLn('ANSWER: ', Sum);
end.
